ENIGMAsystem/SHELL/Graphics_Systems/General/GSbackground.h

```diff
@@ -19,22 +19,23 @@
 #define ENIGMA_GSBACKGROUND_H
 
 #include "Universal_System/scalar.h"
+#include "GScolors.h"
 
 namespace enigma_user
 {
 
 double background_get_texture_width_factor(int backId);
 double background_get_texture_height_factor(int backId);
-void draw_background(int back, gs_scalar x, gs_scalar y, int color = 16777215, gs_scalar alpha = 1);
-void draw_background_stretched(int back, gs_scalar x, gs_scalar y, gs_scalar wid, gs_scalar hei, int color = 16777215, gs_scalar alpha = 1);
-void draw_background_part(int back, gs_scalar left, gs_scalar top, gs_scalar wid, gs_scalar hei, gs_scalar x, gs_scalar y, int color = 16777215, gs_scalar alpha = 1);
-void draw_background_tiled(int back, gs_scalar x, gs_scalar y, int color = 16777215, gs_scalar alpha = 1);
-void draw_background_tiled_area(int back,gs_scalar x, gs_scalar y,gs_scalar x1, gs_scalar y1,float x2,float y2, int color = 16777215, gs_scalar alpha = 1);
-void draw_background_ext(int back, gs_scalar x, gs_scalar y, gs_scalar xscale = 1, gs_scalar yscale = 1, double rot = 0, int color = 16777215, gs_scalar alpha = 1);
+void draw_background(int back, gs_scalar x, gs_scalar y, int color = c_white, gs_scalar alpha = draw_get_alpha());
```

Comment:
>Why are we defaulting the color but not the alpha?  
  
Because that's how it is in GMS, I don't think it's too unreasonable, the color is more accurately defined as 'blend color' rather than 'draw color' when you're dealing with images.  
  
>Either way, do we have compatibility macros? It would be a good idea to #define DEFAULT_IMAGE_ALPHA to either of draw_get_alpha() or 1 atop this file, conditioned on those macros. #undef it at the bottom, too.  
  
Aside from that, how do you do compatibility macros? Are they just #ifndef #define #endif definitions that get overridden at compile if needed?